TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -966,46 +957,68 @@ private void WriteVariantResults()
             foreach (var entry in variantPeptides)
             {
                 var pwsm = entry.BestMatchingPeptides;
-                var nonvariantProtein = false;
+                var variantPeptide = true;
                 foreach (var pep in pwsm)
-                {
-                    if (pep.Peptide.Protein.NonVariantProtein.Name == pep.Peptide.Protein.Accession)
+                {                    
+                    if (pep.Peptide.IsVariantPeptide() == false)
                     {
-                        nonvariantProtein = true;
+                        variantPeptide = false;
                     }
                 }
-                if (nonvariantProtein != true)
+                if (variantPeptide == true)
                 {
                     confidentVariantPeps.Add(entry);
                 }
-            }
-
-            int savCount = 0;
+            }        
+            // count of peptides that contain at least 1 of the given variant type
+            int SNVmissenseCount = 0;
+            int MNVmissenseCount = 0;
             int insertionCount = 0;
             int deletionCount = 0;
             int frameshiftCount = 0;
             int stopGainCount = 0;
-            int stopLossCount = 0;                                 
-            List<PeptideSpectralMatch> modifiedVariantPeptides = confidentVariantPeps.Where(p => p.ModsIdentified.Count() > 0 && p.FdrInfo.QValue <= 0.01 && p.FdrInfo.QValueNotch <= 0.01 && p.IsDecoy == false && p.IsContaminant ==false).ToList(); //modification can be on any AA in variant peptide
+            int stopLossCount = 0;
+
+            // dictionaries facilitate the determination of unique variant sites
+            Dictionary<Protein, HashSet<SequenceVariation>> MNVmissenseVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> SNVmissenseVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> insertionVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> deletionVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> frameshiftVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> stopGainVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();
+            Dictionary<Protein, HashSet<SequenceVariation>> stopLossVariants = new Dictionary<Protein, HashSet<SequenceVariation>>();            
+
+            List<PeptideSpectralMatch> modifiedVariantPeptides = confidentVariantPeps.Where(p => p.ModsIdentified != null && p.ModsIdentified.Count() > 0 && p.FdrInfo.QValue <= 0.01 && p.FdrInfo.QValueNotch <= 0.01 && p.IsDecoy == false && p.IsContaminant ==false).ToList(); //modification can be on any AA in variant peptide
```

Comment:
Check whether you can use the `.Count` property rather than `.Count()`
Consider using `!p.IsDecoy && !p.IsContaminant`